Gems/Atom/RPI/Code/Source/RPI.Public/Shader/Shader.cpp

```diff
@@ -520,6 +532,33 @@ namespace AZ
             return m_asset->GetShaderResourceGroupLayouts(m_supervariantIndex);
         }
 
+        Data::Instance<ShaderResourceGroup> Shader::CreateDrawSrgForShaderVariant(const ShaderOptionGroup& shaderOptions, bool compileTheSrg)
+        {
+            RHI::Ptr<RHI::ShaderResourceGroupLayout> drawSrgLayout = m_asset->GetDrawSrgLayout(GetSupervariantIndex());
+            Data::Instance<ShaderResourceGroup> drawSrg;
```

Comment:
Could this be generalized to support the case where the fallback key is in the PassSrg and not the DrawSrg? Could that be controlled with an input param?
Then you could keep CreateDrawSrgForShaderVariant as a function name for a wrapper function and have this be a more generic CreateSrgForShaderVariant

Reply:
Sure, that sounds reasonable. Is there currently some pass code where the PassSrg has the fallback key that you can point me to?